core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -411,16 +445,10 @@ def node_definition(node, path, cookbook_path)
       node_params = make_node_params(node, @boxes.get_box(box))
       print_node_info(node_params, box)
     end
-    provisioned = !node[1]['product'].nil?
-    if provisioned
-      product = node[1]['product']
-      node_params[:template_path] = product['cnf_template_path'] unless product['cnf_template_path'].nil?
-    else
-      product = { 'name' => 'packages' }
-    end
-    @ui.info("Machine #{node_params[:name]} is provisioned by #{product}")
-    # box with mariadb, maxscale provision - create role
-    role = get_role_description(node_params[:name], product, box)
+    products, cnf_template_path = parse_products_info(node)
```

Comment:
Check whether disablement of Metrics/MethodLength cop is still required for the method.